pkg/querier/queryrange/roundtrip.go

```diff
@@ -269,7 +269,9 @@ func NewSeriesTripperware(
 		queryRangeMiddleware = append(queryRangeMiddleware,
 			queryrange.InstrumentMiddleware("split_by_interval", instrumentMetrics),
 			// The Series API needs to pull one chunk per series to extract the label set, which is much cheaper than iterating through all matching chunks.
-			SplitByIntervalMiddleware(WithSplitByLimits(limits, cfg.SplitQueriesByInterval), codec, splitByTime, splitByMetrics),
+			// Force a 24 hours split by for series API, this will be more efficient with our static daily bucket storage.
+			// This would avoid queriers downloading chunks for same series over and over again for serving smaller queries.
+			SplitByIntervalMiddleware(WithSplitByLimits(limits, 24*time.Hour), codec, splitByTime, splitByMetrics),
```

Comment:
[Q]: I see the only change is hardcoding `24*time.Hour` instead of `cfg.SplitQueriesByInterval`. can't we just use it via config? and mentioned it in the doc? or even default value to 24 hours?

Just wondering what's the rationale behind forcing it.

Reply:
We do the same for labels queries see https://github.com/grafana/loki/blob/fbb8ecaa0669939db1c69ddb45a550c94775df5b/pkg/querier/queryrange/roundtrip.go#L303

The rationale behind forcing it is we have daily buckets in the index so this aligns with it to get the maximum benefit from a single query. Also, the series are not expected to change dramatically every 15 or 30 mins so this lets us avoid a lot of rework.